### PR TITLE
amberol: 0.10.3 -> 2024.1

### PR DESCRIPTION
**Pull request by linsui**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by SigmaSquadron** [comment]
Please split this into three commits:

1. `amberol: migrate to by-name`
2. `amberol: format with nixfmt-rfc-style`
3. `amberol: 0.10.3 -> 2024.1`

File: pkgs/by-name/am/amberol/package.nix [hunk 1]
```diff
@@ -1,40 +1,41 @@
...
-    repo = pname;
+    repo = "amberol";
     rev = version;
-    hash = "sha256-nAoUO0bGToNGD2W8qJmTegrETOJDdM04hI1jjiYkZXI=";
+    hash = "sha256-WuyvTgh9Qc5WcgEssxkytwQpSACd82l5WKeMD0NFOp8=";
   };
 
   cargoDeps = rustPlatform.fetchCargoTarball {
     inherit src;
     name = "${pname}-${version}";
```
Comment (SigmaSquadron) [inline pkgs/by-name/am/amberol/package.nix:37]:
```suggestion
    name = "amberol-${version}";
```

**Review by SigmaSquadron — COMMENTED** [review]
As you're already modernising the package, I'd also recommend removing the `with lib;` from `meta`.

**Review by SigmaSquadron — APPROVED** [review]
You may want to split the update commit into one that just updates the package and another that does the related package modernisation, but that's a nitpick at best.

**Review by SigmaSquadron — APPROVED** [review]
Excellent work!

**Review by momeemt — APPROVED** [review]

**Comment by accelbread** [comment]
Since this update, the background colors from cover art feature seems broken, and the UI is just gray. The flatpak version of Amberol still has the background colors, so I assume there is some missing dependency.

**Comment by accelbread** [comment]
Looking into it further, Amberol 2024.1 requires GTK 4.16 (https://gitlab.gnome.org/World/amberol/-/commit/efa36d3a6a114e9355f3b7c6e2e41a5e21cebf65). Currently gtk4 on master is 4.14.5.

**Pull request merged**